tests/integration_tests.rs

```diff
@@ -747,6 +747,25 @@ fn config_location_when_generating() {
     assert!(tmp_config_path.exists());
 }
 
+#[test]
+fn config_location_from_bat_config_dir_variable() {
+    let bat_conf_dir = tempdir().expect("can't create temporary directory.").path().join("bat_conf_dir/");
+
+    // Create file at BAT_CONFIG_DIR
+    bat_with_config()
+        .env("BAT_CONFIG_DIR", bat_conf_dir.to_str().unwrap())
+        .arg("--generate-config-file")
+        .assert()
+        .success()
+        .stdout(
+            predicate::str::is_match("Success! Config file written to .*config\n")
+                .unwrap()
+        );
```

Comment:
Another idea for a (maybe simpler) test would be similar to `config_read_arguments_from_file`. Wouldn't require a temp. directory. But I guess this is fine as well. Thank you for writing a test!